core/dtn7/src/routing/erouting/processing.rs

```diff
@@ -0,0 +1,255 @@
+            name: eid.node().unwrap(),
+            eid: eid.clone(),
+        }),
+    };
+
+    send_packet(&packet);
+}
+
+fn remove_response_channel(id: &str) {
+    RESPONSES.lock().unwrap().remove(id);
+}
+
+fn create_response_channel(id: &str, tx: oneshot::Sender<Packet>) {
+    RESPONSES.lock().unwrap().insert(id.to_string(), tx);
+}
+
+/// Tries to send a routing requests to the external router and waits for the response.
+/// The wait will be limited to a timeout of 250ms.
+pub async fn sender_for_bundle(bp: &BundlePack) -> (Vec<ClaSenderTask>, bool) {
+    trace!("external sender_for_bundle initiated: {}", bp);
+
+    if CONNECTION.lock().unwrap().is_none() {
+        return (vec![], false);
+    }
+
+    // Register a response channel for the request
+    let (tx, rx) = oneshot::channel();
+    create_response_channel(bp.to_string().as_str(), tx);
+
+    // Send out the SenderForBundle packet
+    let packet: Packet = Packet::SenderForBundle(SenderForBundle {
+        clas: cla_names(),
+        bp: bp.clone(),
+    });
+    send_packet(&packet);
+
+    let res = timeout(time::Duration::from_millis(250), rx).await;
+    if let Ok(Ok(Packet::SenderForBundleResponse(packet))) = res {
+        remove_response_channel(bp.to_string().as_str());
+
+        if packet.bp.to_string() != bp.to_string() {
+            error!("got a wrong bundle pack! {} != {}", bp, packet.bp);
+            return (vec![], false);
+        }
+
+        return (
+            packet
+                .clone()
+                .clas
+                .iter()
+                .filter_map(|sender| {
+                    for cla_instance in &(*CLAS.lock()) {
+                        if sender.agent == cla_instance.name() {
+                            let dest = format!(
+                                "{}:{}",
+                                sender.remote,
+                                sender.port.unwrap_or_else(|| cla_instance.port())
+                            );
+
+                            return Some(ClaSenderTask {
+                                tx: cla_instance.channel(),
+                                dest,
+                                cla_name: cla_instance.name().into(),
+                                next_hop: sender.next_hop.clone(),
+                            });
+                        }
+                    }
+                    None
+                })
+                .collect(),
+            packet.delete_afterwards,
+        );
+    }
+
+    // Signal to the external router that the timeout was reached and no SenderForBundleResponse was processed.
+    // This is needed in case that the response arrived later than the timeout and the connected router thinks
+    // it successfully send its response. Otherwise there is no way for the router to know if its response has
+    // failed.
+    send_packet(&Packet::Timeout(super::Timeout { bp: bp.clone() }));
```

Comment:
cloning the whole bp might be expensive, doing the removing of the response channel first or just cloning the string representation might be cheaper